units/check_blocked_tasks.go

```diff
@@ -84,29 +96,29 @@ func (j *checkBlockedTasksJob) Run(ctx context.Context) {
 			"job":                 j.ID(),
 			"source":              checkBlockedTasks,
 		})
-		return
+		return nil
 	}
 
 	tasksToCheck, err := task.Find(task.ByIds(taskIds))
 	if err != nil {
 		j.AddError(errors.Wrapf(err, "getting tasks to check in distro '%s'", j.DistroId))
-		return
+		return nil
 	}
+	return tasksToCheck
+}
 
-	dependencyCache := map[string]task.Task{}
-	numTasksModified := 0
-	numChecksThatUpdatedTasks := 0
-	for _, t := range tasksToCheck {
-		numModified, err := checkUnmarkedBlockingTasks(&t, dependencyCache)
-		j.AddError(err)
-		numTasksModified += numModified
-		if numTasksModified > 0 {
-			numChecksThatUpdatedTasks++
-		}
+func (j *checkBlockedTasksJob) getTasksToCheckContainer() []task.Task {
+	query := task.IsUndispatchedContainerTasksQuery()
+	query[task.ContainerAllocatedKey] = false
+	tasksToCheck, err := task.FindAll(db.Query(query).Sort([]string{task.ActivatedTimeKey}))
```

Comment:
Does this need to sort on activated time? The sort is needed for the container task queue because that's the only queue-ordering mechanism. But I imagine their order in the queue doesn't matter with regards to dependency checking, unless I'm missing something.

Reply:
Good catch, removed